src/main/java/com/aws/greengrass/logmanager/LogManagerService.java

```diff
@@ -585,25 +601,25 @@ private void processLogsAndUpload() throws InterruptedException {
                             }
                         }
                     }
+                    // If there are no log files for the component, then return.
+                    if (allFiles.isEmpty()) {
```

Comment:
This should not be needed. The only reason you have to do this is because now we are relying on it always having at least 1 because you are using it to get the activeFileName